packages/gestalt/src/Dropdown.js

```diff
@@ -149,8 +160,13 @@ export default function Dropdown({
   onDismiss,
   zIndex,
   maxHeight,
+  mobileOnAnimationEnd,
+  disableMobileUI = true,
```

Comment:
This is an anti-pattern, leading to a confusing DX. For example, which of these yields a disabled mobile UI?
```js
<Dropdown />
<Dropdown disableMobileUI />
<Dropdown disableMobileUI={false} />
```
The first two both do, but how is a developer to know that the first example would disable the mobile UI?

We also plan to make this behavior the default in the future, requiring another breaking change in the future.

We can avoid that future breaking change and encourage all new usages of Dropdown to use the adaptive UI by setting this to our target default of `disableMobileUI = false`, and codemodding all current Dropdown usages.

Reply:
Ryan, 

In my comment above:

<img width="806" alt="Screenshot 2023-05-03 at 9 27 29 PM" src="https://user-images.githubusercontent.com/10593890/236088113-1c8cd991-3844-4e20-9656-0ac6d26d55e5.png">

I'm going to run the codemod, independently from the PR, set all current instances to true, then set default to false

@dangerismycat